### DISCUSSION TITLE
Feature Request: Checklist button in toolbar

### DISCUSSION THREAD

**Discussion by rjonesy** [community/community#17] [Mobile]
This is semi-related to #12 and #16 requests for more buttons too.

**This request is for a checklist button in the toolbar.** We use checklists often as sub-tasks to a larger task, because of the very nice implementation of "3 of 7" in Issues lists show us how many sub-tasks have been completed. It's really great. I noticed you have a bullet-list button in the toolbar which is semi-surprising since that is simply a `-` versus checklist which is more characters `- [ ]`.

As an overall suggestion I would not be afraid to add more buttons to your toolbar. (This is my guessing...but) Github Mobile is a 90% a text editor, and 90% of your users are power user, and the toolbar is incredibly well built – so it can handle it and we can handle it.

**Comment by rnystrom** [comment]
I love this, I think it's a great idea. I just noticed that on the web we have a checkbox button in the markdown editors too.

I like our single bar, but the more options we add the harder it becomes to use. We've also heard reports where people didn't know that you could swipe/scroll the bar to see more actions.

What do you think of Apple's "Notes" app design for controls? Are there any other writing tools you think have great style controls?

**Reply by mbbischoff** [reply]
I agree with @rjonesy below. A horizontal scroll with a one time coaching bounce would work great here.

**Comment by rjonesy** [comment]
Notes is great - best-in-class for that UX type. And Dropbox Paper is best-in-class for its type. Paper started with a bar like yours but you can see they ran into the same "people don't know it scrolls" issue and solved with with an expanding menu. Really well done. 

All that said... I wouldn't do Notes or Paper style in Github. Github has a known quantity of buttons, proven over time, and it's right above me in this desktop editor. 

I would adapt that to mobile (add photo picker for example) and add a horizontal bounce animation to show the user it scrolls, until they interact with it once.
